File: files/200-299/258/CH-258.py
import pandas as pd
import json

path = "200-299/258/CH-258 Column Splitting.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=5)
test = pd.read_excel(path, usecols="D:H", skiprows=1, nrows=5)

result = (
    pd.DataFrame(
        input['Text']
        .str.replace(r"([a-zA-Z0-9\.]+)", r'"\1"', regex=True)
        .map(json.loads)
        .tolist()
    )
    .assign(age=lambda x: x['age'].astype(float))
)

print(result.equals(test))

File: files/200-299/258/CH-258.R
library(tidyverse)
library(readxl)
library(jsonlite)

path = "files/200-299/258/CH-258 Column Splitting.xlsx"
input = read_excel(path, range = "B2:B6")
test  = read_excel(path, range = "D2:H6")

result = input %>%
  mutate(across(everything(),
         ~ str_replace_all(.x, "([a-zA-Z0-9\\.]+)", '"\\1"'))) %>%
  mutate(text_parsed = map(Text, ~ fromJSON(.) %>% as_tibble())) %>%
  unnest(text_parsed) %>% 
  select(-Text) 

all.equal(result, test) 
# > [1] TRUE